plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { TableRow } from '@woocommerce/components/build-types/table/types';
+import { Icon, plugins } from '@wordpress/icons';
+import { gmdateI18n } from '@wordpress/date';
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { Subscription } from '../../types';
+import StatusPopover from './status-popover';
+import ActivationToggle from './activation-toggle';
+import ActionsDropdownMenu from './actions-dropdown-menu';
+
+function getStatus( subscription: Subscription ): {
+	text: string;
+	warning: boolean;
+	explanation?: string;
+} {
+	// TODO add statuses for subscriptions
+	if ( subscription.active ) {
+		return {
+			text: __( 'Active', 'woocommerce' ),
+			warning: false,
+		};
+	}
+
+	if ( subscription.product_key === '' ) {
+		return {
+			text: __( 'Not found', 'woocommerce' ),
+			warning: true,
+			explanation: __( 'This subscription is not found.', 'woocommerce' ),
+		};
+	}
+
+	if ( subscription.expired ) {
+		return {
+			text: __( 'Expired', 'woocommerce' ),
+			warning: true,
+			explanation: __(
+				'To receive updates and support for this extension, you need to purchase a new subscription or consolidate your extensions to one connected account by sharing or transferring this extension to this connected account.',
```

Comment:
This is a bit of a mouthful, how about this?

```suggestion
				'To get updates and support for this extension, you need to purchase a new subscription, or else share or transfer a subscription for this extension from another account.',
```